File: 231-240/238. Product of Array Except Self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let leftMulti = 1;
    let rightMulti = 1;
    let res = [];

    for(let i = 0; i <nums.length;i++){
        res[i] = leftMulti;
        leftMulti *= nums[i];
    }

    for(let i = nums.length-1;i>=0;i--){
        res[i] *= rightMulti;
        rightMulti *= nums[i];
    }
    return res;
};

File: 231-240/238. Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        leftMulti = 1
        rightMulti = 1
        res = [1] * len(nums)

        for i in range(len(nums)):
            res[i] = leftMulti
            leftMulti *= nums[i]

        for i in range(len(nums) - 1, -1, -1):
            res[i] *= rightMulti
            rightMulti *= nums[i]

        return res
